test/testdrive/linked-clusters.td

```diff
@@ -0,0 +1,305 @@
+> ALTER SINK snk2 SET (SIZE = '4')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+materialize_public_snk2  linked  4
+> ALTER SINK snk2 RESET (SIZE)
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+materialize_public_snk2  linked  ${arg.default-storage-size}
+
+# Test that only the default clusters and replicas remain after dropping
+# the sinks.
+> DROP SINK snk1
+> DROP SINK snk2
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
+
+# Test that two objects with the same name create distinct clusters
+# as the schema name is included in the cluster name
+
+> CREATE SCHEMA schema1;
+> CREATE SCHEMA schema2;
+
+> CREATE SOURCE schema1.s1 FROM LOAD GENERATOR COUNTER
+> CREATE SINK schema2.s1 FROM v
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'snk-${testdrive.seed}')
+  FORMAT JSON ENVELOPE DEBEZIUM
+
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_schema1_s1
+materialize_schema2_s1
+
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
+materialize_schema1_s1  linked  ${arg.default-storage-size}
+materialize_schema2_s1  linked  ${arg.default-storage-size}
+
+> DROP SCHEMA schema1 CASCADE;
+> DROP SCHEMA schema2 CASCADE;
+
+# Test capitalizarion behavior
+
+> CREATE SOURCE "s1" FROM LOAD GENERATOR COUNTER
+> CREATE SINK "S1" FROM v
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'snk-${testdrive.seed}')
+  FORMAT JSON ENVELOPE DEBEZIUM
+
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_s1
+materialize_public_S1
+
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
+materialize_public_s1  linked  ${arg.default-storage-size}
+materialize_public_S1  linked  ${arg.default-storage-size}
+
+> DROP SOURCE "s1";
+> DROP SINK "S1";
+
+# Test that an already-existing cluster is taken into account when
+# selecting a name for the linked cluster
+
+> CREATE CLUSTER materialize_public_s1 REPLICAS (r1 (SIZE '1'));
+> CREATE SOURCE s1 FROM LOAD GENERATOR COUNTER
+
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_s1
+materialize_public_s11
+
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
+materialize_public_s1  r1  1
+materialize_public_s11 linked ${arg.default-storage-size}
+
+> DROP CLUSTER materialize_public_s1;
+> DROP SOURCE s1;
```

Comment:
One more useful test might be to have a source and sink with the same name and ensure that the cluster names are unique.

Reply:
Oh, I meant the same schema but didn't realize we had that restriction. You can ignore this.